pkg/reader/ccip.go

```diff
@@ -582,6 +585,23 @@ func (r *ccipChainReader) getSourceChainsConfig(
 			if err != nil {
 				return fmt.Errorf("failed to get source chain config: %w", err)
 			}
+
+			// This may happen due to some sort of regression in the codec that unmarshals
+			// chain data -> go struct.
+			if len(resp.OnRamp) == 0 {
```

Comment:
I think it makes more sense to return an error here than invalid data. Lmk your thoughts @winder

Reply:
Yes, let's detect problems and generate the errors early. Full address validation could happen in `DiscoverContracts`

Reply:
Good note about address validation, I think here you can put the address validation, i.e if the source chain is EVM and the # of bytes is not what is expected, error.